ml-agents/mlagents/trainers/trainer/rl_trainer.py

```diff
@@ -40,7 +33,8 @@ def __init__(self, *args, **kwargs):
         }
         self.update_buffer: AgentBuffer = AgentBuffer()
         self._stats_reporter.add_property(
-            StatsPropertyType.HYPERPARAMETERS, self.trainer_parameters
+            StatsPropertyType.HYPERPARAMETERS,
+            cattr.unstructure(self.trainer_parameters),
```

Comment:
Minor but this feels like it leaks the abstraction of the `TrainerSettings` implementation.  Perhaps this could be implemented as `TrainerSettings.as_dict`?

Reply:
Made a mixin that defines the `as_dict` method and gave it to TrainerSettings and RunOptions - should clean up the code in learn.py a bit.